dalite_xblock/dalite_xblock.py

```diff
@@ -156,14 +190,74 @@ def student_view(self, context):
         :returns: XBlock HTML fragment
         :rtype: xblock.fragment.Fragment
         """
-        if not self.is_lti_ready:
-            fragment = Fragment()
-            context.update(self._get_context_for_template())
-            fragment.add_content(loader.render_django_template('/templates/xblock_data_not_filled.html', context))
-            return fragment
-        fragment = super(DaliteXBlock, self).student_view(context)
-        fragment.add_javascript(loader.load_unicode('public/js/dalite_xblock.js'))
-        fragment.initialize_js('DaliteXBlock')
+        return self._render_student_view(context, False)
+
+    @contextlib.contextmanager
+    def add_extra_custom_params(self, additional_custom_parameters):
+        """
+        Temporarily adds custom parameters to this xblock.
+
+        These parameters will be sent to the `lti` provider by any calls made during that time.
+
+        :param list additional_custom_parameters: A list of parameters in a 'key=value` format.
+               Eg: ``[u'action=launch-admin']``
+        """
+        current_params = self.custom_parameters
+        try:
+            self.custom_parameters = list(self.custom_parameters) + additional_custom_parameters
+            yield
+        finally:
```

Comment:
Is the list addition operation likely to fail?  If not, then why wrap it in a `try`/`finally`?